kvdb/interface.go

```diff
@@ -135,6 +135,86 @@ type RwCursor = walletdb.ReadWriteCursor
 // instead.
 type RwTx = walletdb.ReadWriteTx
 
+// ExtendedRTx is an extension to walletdb.ReadTx to allow prefetching of keys.
+type ExtendedRTx interface {
+	RTx
+
+	// RootBucket returns the "root bucket" which is pseudo bucket used
+	// when prefetching (keys from) top level buckets.
+	RootBucket() RBucket
+}
+
+// ExtendedRBucket is an extension to walletdb.ReadBucket to allow prefetching
+// of different kind of keys.
+type ExtendedRBucket interface {
+	RBucket
+
+	// Prefetch will prefetch keys (exact match) and ranges (prefix match).
+	// So that subsequent fetches for those keys and keys in ranges don't
+	// go the the DB.
+	Prefetch(keys []string, ranges []string)
```

Comment:
To me this seems to be an interface that is pretty etcd-specific. Isn't it possible to define a more abstract interface for prefetching at the `kvdb` level?